src/ape_ethereum/transactions.py

```diff
@@ -61,14 +61,18 @@ def serialize_transaction(self) -> bytes:
         signature = (self.signature.v, to_int(self.signature.r), to_int(self.signature.s))
         signed_txn = encode_transaction(unsigned_txn, signature)
 
-        if self.sender and EthAccount.recover_transaction(signed_txn) != self.sender:
+        if (
+            self.sender
+            and EthAccount.recover_transaction(signed_txn) != self.sender
+            and not ignore_sign_match
```

Comment:
If you put `and txn.signature` prior to the previous leg of the conditional, you can emulate this same thing without the kwarg

Reply:
this is another thing, if you use impersonated account then traverse block transactions, you will see there is a signature for the transaction which to me really weird.

Reply:
i tried to find the account address who signed this but never found the address coming in with tx.

Reply:
so self.signature is not doing that

Reply:
it would also fail at line 56 otherwise